src/test/java/poll/service/VoteServiceTest.java

```diff
@@ -1,4 +1,142 @@
+                new Vote(3L, "Ivan Zaitsev", LocalDateTime.of(2021, 1, 2, 9, 30), Answer.YES),
+                new Vote(4L, "Osmany Juantorena", LocalDateTime.of(2021, 1, 2, 9, 45), Answer.NO),
+                new Vote(5L, "Uros Kovacevic", LocalDateTime.of(2021, 1, 2, 10, 0), Answer.NO),
+                new Vote(6L, "Marko Podrascanin", LocalDateTime.of(2021, 1, 2, 14, 30), Answer.NO),
+                new Vote(7L, "Swan Ngapeth", LocalDateTime.of(2021, 1, 3, 8, 0), Answer.YES),
+                new Vote(8L, "Kamil Semeniuk", LocalDateTime.of(2021, 1, 3, 14, 0), Answer.YES),
+                new Vote(9L, "Seyed Mousavi", LocalDateTime.of(2021, 1, 3, 14, 59), Answer.YES)
+        );
+    }
+
+
+    @Test
+    void should_find_vote_by_id() {
+        //given
+        long id = 1L;
+        when(voteRepository.findById(id)).thenReturn(Optional.of(answersDb.get(0)));
+        //when
+        //then
+        assertThat(answersDb.get(0)).isEqualTo(voteService.getVoteById(id));
+    }
+
+    @Test
+    void should_throw_exception_when_findById_not_found_Vote() {
+        //given
+        long id = 100;
+        when(voteRepository.findById(id)).thenReturn(Optional.empty());
+        //when
+        //then
+        final var thrownException = catchThrowable(() -> voteService.getVoteById(id));
+        assertThat(thrownException)
+                .isInstanceOf(VoteNotFoundException.class)
+                .hasMessageContaining("Vote with id " + id + " not found");
+    }
+
+    @Test
+    void should_create_new_vote_when_not_found_by_id() {
+        //given
+        long id = 100;
+        when(voteRepository.findById(id)).thenReturn(Optional.empty());
+        //when
+        final var result = voteService.getOrCreateVote(id);
+        //then
+        assertThat(result.getId()).isEqualTo(id);
+        assertThat(result.getName()).isNull();
+        assertThat(result.getDate()).isNull();
+        assertThat(result.getAnswer()).isNull();
+    }
+
+
+    @Test
+    void should_return_all_yes_votes_by_name() {
+        //given
+        String name = "Ngapeth";
+        when(voteRepository.findByName(name)).thenReturn(List.of(answersDb.get(6), answersDb.get(0)));
+        //when
+        final var result = voteService.getAllYesVotesByName(name);
+        //then
+        assertThat(result).hasSize(1);
+        assertThat(result.get(0).getName()).contains(name);
+        assertThat(result.get(0).getAnswer()).isEqualTo(Answer.YES);
+        assertThat(List.of(answersDb.get(6))).isEqualTo(voteService.getAllYesVotesByName(name));
+    }
+
+    @Test
+    void should_return_empty_list_when_name_not_exist() {
+        //given
+        String name = "Miecio";
+        when(voteRepository.findByName(name)).thenReturn(Collections.emptyList());
+        //when
+        //then
+        assertThat(voteService.getAllYesVotesByName(name)).isEmpty();
+    }
+
+    @Test
+    void should_return_yes_answer_won() {
```

Comment:
We could specify in the test name *when we return the yes*. Sth like: ..._when_majority_answers_...